pkgs/applications/misc/xmrig/xmrig-cuda.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv
+, autoAddOpenGLRunpathHook
+, fetchFromGitHub
+, cmake
+, pkgs
+, tree
+, cudaArch ? 61
+}:
+
+let
+  libcudaDir = "/run/opengl-driver/lib";
+in
+stdenv.mkDerivation rec {
+  inherit (pkgs.cudaPackages) cudatoolkit cuda_cudart;
```

Comment:
We're trying not to use `cudaPackages.cudatoolkit` if we can avoid it. It's the old `.run`-file based distribution of cudatoolkit. What we try to use now is the redistributable packages that nvidia publishes at https://developer.download.nvidia.com/compute/cuda/redist/

For example, `cuda_cudart` corresponds to https://developer.download.nvidia.com/compute/cuda/redist/cuda_cudart/

All of the attributes in `cudaPackages` that begin with `cuda_` or `lib` are redist packages, e.g.: `cuda_nvcc`, `cuda_nvrtc`, `libcublas`, etc.

And there's another copy of `libcudart.so` in the old `cudatoolkit`, which we don't want to mix.

Also we would normally move this `inherit` up into the `let ... in` expression, because otherwise these attributes get propagated into the derivation as environment variables, and we just don't need that


TLDR: Let's try and remove `cudatoolkit`